src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -262,10 +266,43 @@ public List<LfLimit> getLimits2(final LimitType type) {
     }
 
     @Override
-    public void updateState(LfNetworkStateUpdateParameters parameters) {
+    public void updateState(LfNetworkStateUpdateParameters parameters, LfNetworkUpdateReport updateReport) {
         var branch = getBranch();
 
-        updateFlows(p1.eval(), q1.eval(), p2.eval(), q2.eval());
+        if (isDisabled()) {
+            updateFlows(Double.NaN, Double.NaN, Double.NaN, Double.NaN);
+        } else {
+            updateFlows(p1.eval(), q1.eval(), p2.eval(), q2.eval());
+        }
+
+        // in case of automation system simulation we might need to update IIDM terminals connection status
+        if (parameters.isSimulateAutomationSystems()) {
+            boolean connectedSide1Before = branch.getTerminal1().isConnected();
+            boolean connectedSide2Before = branch.getTerminal2().isConnected();
+            boolean connectedSide1After = !isDisabled() && isConnectedSide1();
```

Comment:
Here add a comment maybe because `isConnectedSide1()` and `isConnectedSide2()` is not needed for the moment?